subprojects/distributions/src/integTest/groovy/org/gradle/BinaryBreakageIntegrationTest.groovy

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle
+
+import japicmp.cmp.JApiCmpArchive
+import japicmp.cmp.JarArchiveComparator
+import japicmp.cmp.JarArchiveComparatorOptions
+import japicmp.config.Options
+import japicmp.model.JApiClass
+import japicmp.output.stdout.StdoutOutputGenerator
+import org.gradle.api.internal.file.pattern.PathMatcher
+import org.gradle.api.internal.file.pattern.PatternMatcherFactory
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.executer.GradleDistribution
+import org.gradle.integtests.fixtures.executer.UnderDevelopmentGradleDistribution
+import org.gradle.integtests.fixtures.versions.ReleasedVersionDistributions
+import spock.lang.Unroll
+
+import static org.junit.Assume.assumeTrue
+
+@Unroll
+class BinaryBreakageIntegrationTest extends AbstractIntegrationSpec{
```

Comment:
Currently, I would leave it as an integration test for a few reasons:
- We use our integration test infrastructure do download, unpack and locate the previous/current Gradle distribution
- The test takes 6 seconds which is pretty fast. The slowest test in that integ test source set is actually running a Gradle build on the source distribution to see if it is valid
- Reporting and traceability of test failures is already included by it being a test.

We can still decide to change it to a different test task (e.g. its own test task) or its own task type in the future.